### PR TITLE
Add deprecated tags to API summaries

### PR DESCRIPTION
**Pull request by calebmshafer**
Update the extract-api summary to include potentially two release tags as a single API can be both in a deprecated state in addition to alpha/beta/public (and technically internal but that doesn't make much sense...).  This change should help give a better overview of the state of our APIs.

UI Framework API change, I noticed that WidgetState enum had deprecated setup incorrectly.  Switched it to the correct syntax, please let me know if that deprecated flag was mistakenly added.

All other API changes are a by-product of the change to include both deprecated and other tags.

### CODE REVIEW

File: ui/framework/src/test/frontstage/FrontstageDef.test.tsx [hunk 1]
```diff
@@ -187,15 +187,15 @@ describe("float and dock widget", () => {
...
-    frontstageDef.floatWidget("t1", {x:55, y:105});
+    frontstageDef.floatWidget("t1", { x: 55, y: 105 });
     nineZoneStateSetter.calledOnce.should.true;
 
     frontstageDef.floatWidget("ta");
     nineZoneStateSetter.calledOnce.should.true;
   });
 
   it("should not float if v1", () => {
-    let state = createNineZoneState({size: {height: 1000, width: 1600}});
```
Comment (bsteinbk) [inline ui/framework/src/test/frontstage/FrontstageDef.test.tsx:198]: @calebmshafer - Do you have some extension loaded that adds spaces?  If not, why would eslint add them for you and not for us? The reason I ask is sometimes I feel VS Code is not consistently adding spaces on file save and I am not sure why.

Comment (calebmshafer) [reply ui/framework/src/test/frontstage/FrontstageDef.test.tsx:198]: Possibly an Extension is missing from our recommended settings. I'll take a pass now but I do have 20+ extensions installed :)

Do you have all recommended Extensions installed?  `.vscode/extensions.json`

Comment (bsteinbk) [reply ui/framework/src/test/frontstage/FrontstageDef.test.tsx:198]: Yep, Got all the recommended ones. It does seem to format spaces properly at least 90% of time, so I should be happy about that.

**Review by bsteinbk — APPROVED** [review]

File: ui/framework/src/test/frontstage/FrontstageDef.test.tsx (hunk 1 above)
Comment (calebmshafer) [reply ui/framework/src/test/frontstage/FrontstageDef.test.tsx:198]: Sounds good. I looked through my VSCode settings and I don't see anything that should effect this. The only difference most likely is I'm running on Mac vs. Windows.  I'll see if it auto-formats for me on Windows in this case.

**Comment by grigasp** [comment]
Not sure if it's possible, but wouldn't it be more convenient if there was one line per API, e.g. `public;deprecated;MyApi`?

**Review by pmconne — APPROVED** [review]

**Comment by calebmshafer** [comment]
> Not sure if it's possible, but wouldn't it be more convenient if there was one line per API, e.g. `public;deprecated;MyApi`?

@grigasp I could do it that way as well. I found this way easier to filter and graph in Excel rather than having it all on the same line.  However, if it makes it easier to read when it's directly in csv I can switch it.

**Comment by grigasp** [comment]
> > Not sure if it's possible, but wouldn't it be more convenient if there was one line per API, e.g. `public;deprecated;MyApi`?
> 
> @grigasp I could do it that way as well. I found this way easier to filter and graph in Excel rather than having it all on the same line. However, if it makes it easier to read when it's directly in csv I can switch it.

It's my personal preference to see them in one line, but I don't mind them being two lines if that's useful somewhere.

**Review by grigasp — APPROVED** [review]

**Review by wgoehrig — APPROVED** [review]

**Review by Animagne — APPROVED** [review]

**Review by ramanujam-raman — APPROVED** [review]

File: test-apps/ui-test-app/src/frontend/appui/frontstages/NestedFrontstage1.tsx [hunk 2]
```diff
@@ -129,7 +130,7 @@ class FrontstageToolWidget extends React.Component {
 
   public render() {
     return (
-      <ToolWidget
+      <ToolWidget // eslint-disable-line deprecation/deprecation
```
Comment (calebmshafer) [inline test-apps/ui-test-app/src/frontend/appui/frontstages/NestedFrontstage1.tsx:133]: @bsteinbk @DanEastBentley I didn't fix this but this should probably be using.

**Review by mgooding — APPROVED** [review]

**Review by scsewall — APPROVED** [review]

**Pull request merged**